internal/operators/cnv/cnv_operator.go

```diff
@@ -0,0 +1,117 @@
+package cnv
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/hardware"
+	"github.com/openshift/assisted-service/internal/operators/api"
+	models "github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+type operator struct {
+	log logrus.FieldLogger
+}
+
+// NewCNVOperator creates new instance of a Container Native Virtualization installation plugin
+func NewCNVOperator(log logrus.FieldLogger) *operator {
+	return &operator{
+		log: log,
+	}
+}
+
+// GetType reports type of an operator this Operator manages
+func (o *operator) GetType() models.OperatorType {
+	return models.OperatorTypeCnv
+}
+
+// GetDependencies provides a list of dependencies of the Operator
+func (o *operator) GetDependencies() []models.OperatorType {
+	return []models.OperatorType{models.OperatorTypeLso}
+}
+
+// GetClusterValidationID returns cluster validation ID for the Operator
+func (o *operator) GetClusterValidationID() string {
+	return string(models.ClusterValidationIDCnvRequirementsSatisfied)
+}
+
+// GetHostValidationID returns host validation ID for the Operator
+func (o *operator) GetHostValidationID() string {
+	return string(models.HostValidationIDCnvRequirementsSatisfied)
+}
+
+// ValidateCluster always return "valid" result
+func (o *operator) ValidateCluster(ctx context.Context, cluster *common.Cluster) (api.ValidationResult, error) {
+	// No need to validate cluster because it will be validate on per host basis
+	return api.ValidationResult{Status: api.Success, ValidationId: o.GetClusterValidationID()}, nil
+}
+
+// ValidateHost always return "valid" result
+func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, host *models.Host) (api.ValidationResult, error) {
+	var inventory models.Inventory
+	if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
+		o.log.Errorf("Failed to get inventory from host with id %s", host.ID)
+		return api.ValidationResult{Status: api.Failure}, err
+	}
+	// TODO: validate available devices on worker node like gpu and sr-iov and check whether there is enough memory to support them
+	if host.Role == models.HostRoleWorker {
+		cpu, _ := o.GetCPURequirementForWorker(ctx, cluster)
+		if inventory.CPU.Count < cpu {
+			return api.ValidationResult{Status: api.Failure, Reasons: []string{"Insufficient CPU to deploy CNV. Required CPU count is: " + strconv.FormatInt(inventory.CPU.Count, 10)}}, nil
```

Comment:
Shouldn't you be using `cpu` in the reasons instead of the `inventory.CPU.Count`? The later is what is already available, while you seem to want to express the amount that is at least expected, thus `cpu` should be the number to use here.

Reply:
correct, will fix